test/unit/test_diagram.py

```diff
@@ -46,6 +48,17 @@ def test_job_agg_data():
     assert job_agg.number_of_columns() == 6
 
 
+def test_seq():
+    """
+    function to test the Class JobData inside autosubmit/monitor/diagram.py
+    """
+    for value in diagram._seq(100, 2, 10):
+        assert value is not None
+        assert isinstance(value, int)
+
+def test_populate_statistics(jobs):
+    """ function to test the Class JobData inside autosubmit/monitor/diagram.py """
```

Comment:
This function is now empty? (I checked out the branch as this wasn't a draft PR :grimacing: